### DISCUSSION TITLE
How to change the focus color of button

### DISCUSSION THREAD

**Discussion by appledino3604** [prompt-toolkit/python-prompt-toolkit#1902] [Q&A]
I want to change the color focus when a button is focused, i looked at the buttons.py  in the examples folder. But it does not work, the only color which is appplied to the focus bar is red, setting anything else in the style sheet does not change its color (remains red). Please help

**Comment by TayfurYldz** [comment]
The focused button uses a separate style class. In the current widget implementation the `Window` returns `class:button.focused` when the button has focus, and the built-in default for that class is a red background.

So override `button.focused` in the `Style` that is actually passed to your `Application` / `Dialog`, for example:

```python
from prompt_toolkit.styles import Style

style = Style.from_dict({
    "button": "bg:#444444 #ffffff",
    "button.focused": "bg:#005f87 #ffffff",
})
```

If changing `button` alone appears to work until focus and then turns red, that's expected: the focused state switches to the more specific `button.focused` class.

The repository's button example uses the same selector (`"button focused"` in the list-style syntax), and the default stylesheet defines `button.focused` as red. If an override still doesn't show up, the next thing to check is that this `style` object is the one supplied to the top-level application/dialog rather than a style object that is never attached.
